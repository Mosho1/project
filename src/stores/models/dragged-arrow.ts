import { types } from 'mobx-state-tree';
import { calculateBezierPoints } from '../utils/utils';

export const DraggedArrow = types.model('DraggedArrow', {
    startX: types.number,
    startY: types.number,
    endX: types.number,
    endY: types.number,
}).views(self => {
    return {
        get points() {
            return calculateBezierPoints(
                self.startX,
                self.startY,
                self.endX,
                self.endY,
            );
        }
    };
}).actions(self => {
    const start = (x: number, y: number) => {
        self.startX = x;
        self.startY = y;
        return self;
    };

    const end = (x: number, y: number) => {
        self.endX = x;
        self.endY = y;
        return self;
    };

    return {
        start, end
    };
});

type IDraggedArrowType = typeof DraggedArrow.Type;
export interface IDraggedArrow extends IDraggedArrowType {};