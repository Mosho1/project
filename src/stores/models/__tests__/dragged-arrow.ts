import { DraggedArrow, IDraggedArrow } from "../dragged-arrow";

let a: IDraggedArrow
beforeEach(() => {
    a = DraggedArrow.create({ startX: 0, startY: 0, endX: 50, endY: 50 });
});

test('points', () => {
    expect(a.points).toMatchSnapshot();
});

test('start', () => {
    expect(a.start(1, 2)).toMatchSnapshot();
    expect(a.start(3, 4)).toMatchSnapshot();
});

test('end', () => {
    expect(a.end(1, 2)).toMatchSnapshot();
    expect(a.end(3, 4)).toMatchSnapshot();
});
