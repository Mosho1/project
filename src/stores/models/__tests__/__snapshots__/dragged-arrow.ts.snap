// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end 1`] = `
Object {
  "endX": 1,
  "endY": 2,
  "startX": 0,
  "startY": 0,
}
`;

exports[`end 2`] = `
Object {
  "endX": 3,
  "endY": 4,
  "startX": 0,
  "startY": 0,
}
`;

exports[`points 1`] = `
Array [
  0,
  0,
  50,
  0,
  0,
  50,
  50,
  50,
]
`;

exports[`start 1`] = `
Object {
  "endX": 50,
  "endY": 50,
  "startX": 1,
  "startY": 2,
}
`;

exports[`start 2`] = `
Object {
  "endX": 50,
  "endY": 50,
  "startX": 3,
  "startY": 4,
}
`;
